scripts-available/CDB_ForeignTable.sql

```diff
@@ -139,6 +139,123 @@ $$
 LANGUAGE plpgsql VOLATILE PARALLEL UNSAFE;
 
 
+-- A function to set up a user-defined foreign data server
+-- It does not read from CDB_Conf
+--
+-- Sample call:
+-- SELECT cartodb.CDB_SetUp_User_Foreign_Server('amazon', '{
+--    "server": {
+--      "extensions": "postgis",
+--      "dbname": "testdb",
+--      "host": "myhostname.us-east-2.rds.amazonaws.com",
+--      "port": "5432"
+--    },
+--    "user_mapping": {
+--      "user": "fdw_user",
+--      "password": "secret"
+--    }
+-- }');
+--
+-- Underneath it will:
+--   * Set up postgresql_fdw
+--   * Create a server with the name 'amazon'
+--   * Create a role called 'amazon' to manage access
+--   * Create a user mapping with that role 'amazon'
+--   * Create a schema 'amazon' as a convenience to set up all foreign
+--     tables over there
+--
+-- It is the responsibility of the caller to grant that role to either:
+--   * Nobody
+--   * Specific roles: GRANT amazon TO role_name;
+--   * Members of the organization: SELECT cartodb.CDB_Grant_Role_To_Org_Members('amazon'); TODO
+--   * The publicuser: GRANT amazon TO publicuser;
+CREATE OR REPLACE FUNCTION @extschema@.CDB_SetUp_User_Foreign_Server(fdw_name NAME, config json)
```

Comment:
Do you think it's worth to have different names for `schema/user/server`? I mean, from the point of view of management having different objects with the very same name cloud mislead, couldn't it?

Reply:
I applied "convention over configuration" which IMO is the only way to simplify things. The alternative is what we did with the "operations manual", which is to expose the whole PG SQL syntax and require a superadmin to setup things.

They have the same name, but they can be listed in different views and are in different tables in the PG catalog, so I don't really see drawbacks with this approach.

That said, there's some middle ground which is to [prefix/suffix with types](https://softwareengineering.stackexchange.com/questions/14789/what-is-the-appeal-of-systems-hungarian) as the "convention", but I wouldn't do it as we don't really need to carry domain-type info with the names of different entities in PG.

Reply:
👌